### PR TITLE
Add iOS color & gradient Swift generation

### PR DESCRIPTION
**Pull request by kylehickinson**
This adds Swift code generation for the leo design system colours & gradients 

This will also resolve https://github.com/brave/leo/issues/378 as it includes the code to generate start/stop points for CSS angle-based gradients.

### CODE REVIEW

File: src/tokens/transformation/ios/colorsets.js [hunk 1]
```diff
@@ -45,8 +45,16 @@ module.exports = {
...
       .forEach((token) => {
-        const colorsetPath = `${assetPath}/${changeCase.pascalCase(
-          token.path.slice(2).join(' ')
+        if (token.name === 'ColorBlack' || token.name === 'ColorWhite') {
+          // The system already provides these and will conflict with the names
+          return
+        }
+        const colorsetPath = `${assetPath}/${changeCase.snakeCase(
+          token.path
+            .filter((i) => {
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/ios/colorsets.js:54]: Optional: 
```suggestion
            .filter((i) => i !== 'dark' && i !== 'light' && i !== 'color')
```

we should change the `!=` to `!==` though, even if you don't move to an inline arrow function

File: src/tokens/transformation/ios/index.js [hunk 2]
```diff
@@ -6,3 +11,61 @@ module.exports = {
...
+})
+
+StyleDictionary.registerFormat({
+  name: 'ios/colorSetAccessors',
+  formatter: ({ dictionary, options, file }) => {
+    const template = _template(
+      fs.readFileSync(__dirname + '/templates/ColorSetAccessors.swift.template')
+    )
+
+    var colors = {}
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/ios/index.js:46]: Nit:
```suggestion
    const colors = {}
```

File: src/tokens/transformation/ios/index.js [hunk 3]
```diff
@@ -6,3 +11,61 @@ module.exports = {
...
+      .filter((token) => token.type === 'color')
+      .forEach((token) => {
+        if (token.name === 'ColorBlack' || token.name === 'ColorWhite') {
+          // The system already provides these and will conflict with the names
+          return
+        }
+        const name = changeCase.snakeCase(
+          token.path
+            .filter((i) => {
+              return i != 'dark' && i != 'light' && i != 'color'
```
Comment (fallaciousreasoning) [inline src/tokens/transformation/ios/index.js:57]: Nit: `!=` ==> `!==`

Optional: convert to inline arrow function

**Review by fallaciousreasoning — APPROVED** [review]
LGTM @kylehickinson! Thanks for doing this, it looks great!

File: src/tokens/transformation/ios/colorsets.js (hunk 1 above)
Comment (kylehickinson) [reply src/tokens/transformation/ios/colorsets.js]: Fixed! Though the formatter forced it to include the parenthesis for the arg and the conditions onto a separate line

File: src/tokens/transformation/ios/index.js (hunk 2 above)
Comment (kylehickinson) [reply src/tokens/transformation/ios/index.js]: Done!

File: src/tokens/transformation/ios/index.js (hunk 3 above)
Comment (kylehickinson) [reply src/tokens/transformation/ios/index.js]: Fixed! Though the formatter forced it to include the parenthesis for the arg and the conditions onto a separate line

**Comment by fallaciousreasoning** [comment]
Does this resolve https://github.com/brave/leo/issues/377? Or is that a whole different kettle of fish :D

**Comment by kylehickinson** [comment]
It solves it for iOS in terms of emulating CSS but it doesn't solve for the lack of start/end points defined in Figma, so that we can always render the same as the design files 🙂

**Pull request merged**